source/lib/vagrant-openstack-provider/version_checker.rb

```diff
@@ -0,0 +1,74 @@
+require 'colorize'
+require 'singleton'
+require 'vagrant-openstack-provider/version'
+
+module VagrantPlugins
+  module Openstack
+    class VersionChecker
+      include Singleton
+
+      #
+      # :latest, :outdated or :unstable
+      #
+      # A version is considered unstable if it does not
+      # respect the pattern or if it is greater than the
+      # latest from rubygem
+      #
+      attr_accessor :status
+
+      def initialize
+        @status = nil
+      end
+
+      #
+      # Check the latest version from rubygem and set the status
+      #
+      def check
```

Comment:
This method should have a global timeout, of preference very short.

We should also add a global rescue, and just do nothing if anything fails. For instance if rubygems is not available, if we are running the provider on a private corporate network, this check should not degrade the user experience